### PR TITLE
Remove unnecessary dependencies

### PR DESCRIPTION
**Pull request by sungam3r**
image
I think that these dependencies were added unintentionally due to sample projects that reference the main package.

In fact, only two are enough:
image

### CODE REVIEW

**Comment by nblumhardt** [comment]
Thanks Ivan! This is #110 - moving this package towards a "batteries included" approach, since the core `UseSerilog()` functionality is now implemented in the (low dependency) _Serilog.Extensions.Hosting_ package.

It would be interesting to know which features of _Serilog.AspNetCore_ you're using, so that we can evaluate whether this is the optimal factoring. Cheers!

**Comment by sungam3r** [comment]
Wow! I could not even imagine that it was intended. I highly recommend not including such dependencies. It's like shooting yourself in the foot. With the same success, you can turn on a bunch of other sinks/formatters/appconfigs/etc. in the hope that they will be useful to someone. But in fact, only version conflicts will arise when the consumer uses a package of another major version. These are all spurious dependencies.

> The goal is to make the onboarding process smoother.

In this case, you can make a separate package and name it like `Serilog.AspNetCore.All` but I still would not recommend doing so.

For some consumers, this concern actually turns into problems of versioning dependencies. And the other part simply does not use this "ballast". You have already done sample projects with good examples. No need to mix.

**Comment by sungam3r** [comment]
> It would be interesting to know which features of Serilog.AspNetCore you're using, so that we can evaluate whether this is the optimal factoring.

I use `Serilog.Sinks.Kafka` for example.

**Comment by nblumhardt** [comment]
Thanks for the reply!

As you'd know, the Serilog packages are all very fine-grained, for many of the reasons you point out. Unfortunately, something resembling a meta-package is needed if we're going to meet the ease-of-setup expectations set by _Microsoft.Extensions.Logging_/ASP.NET Core, which includes console and debug logging by default. 

> I use Serilog.Sinks.Kafka for example.

Sorry, I meant which features _from this package_? Since the core logging adapters and `UseSerilog()` are now in _Serilog.Extensions.Logging_ and _Serilog.Extensions.Hosting_, the only feature actually implemented directly in this package is `UseSerilogRequestLogging()` - unless you're using that middleware, then _Serilog.Extensions.Hosting_ is possibly the dependency-free package that you're looking for. Let me know! 👍 

None of this set in stone, of course! It will be interesting to see whether the batteries-included approach here reduces some of the support around this package, or ends up increasing it. We can definitely re-evaluate and change direction if needed.

File: src/Serilog.AspNetCore/Serilog.AspNetCore.csproj [hunk 1]
```diff
@@ -21,16 +21,8 @@
...
-    <PackageReference Include="Serilog" Version="2.8.0" />
     <PackageReference Include="Serilog.Extensions.Hosting" Version="3.0.0" />
-    <PackageReference Include="Serilog.Sinks.Console" Version="3.1.1" />
-    <PackageReference Include="Serilog.Sinks.File" Version="4.0.0" />
-    <PackageReference Include="Serilog.Sinks.Debug" Version="1.0.1" />
-    <PackageReference Include="Serilog.Settings.Configuration" Version="3.1.0" />
-    <PackageReference Include="Serilog.Formatting.Compact" Version="1.0.0" />
     <PackageReference Include="Microsoft.AspNetCore.Hosting.Abstractions" Version="2.0.0" />
-    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="2.0.0" />
-    <PackageReference Include="Microsoft.Extensions.Logging" Version="2.0.0" />
```
Comment (nblumhardt) [inline src/Serilog.AspNetCore/Serilog.AspNetCore.csproj]: We should in any case be able to get rid of these two 👍

File: README.md [hunk 2]
```diff
@@ -205,7 +205,7 @@ Finally, pass the provider collection into `UseSerilog()`:
 
 Providers registered in _Startup.cs_ with `AddLogging()` will then receive events from Serilog.
 
-**Using iniline initialization:**
+**Using inline initialization:**
```
Comment (nblumhardt) [inline README.md]: Argh, nice catch 😅

**Comment by sungam3r** [comment]
> unless you're using that middleware, then Serilog.Extensions.Hosting is possibly the dependency-free package that you're looking for

When I read this comment, I realized that this is what I needed. But when I opened the source code, I was disappointed. It uses `IHostBuilder`, but I need `IWebHostBuilder`. So `Serilog.Extensions.Hosting` package only integrates logging to Generic Host and the package name can be misleading.

Relates to #126

**Comment by nblumhardt** [comment]
Ah, thanks for the follow-up. This is a point-in-time issue, I think, since in .NET Core 3.0, startup will be `HostBuider`-based (with `WebHostBuilder` deprecated), so generally the `IHostBuilder` will be the desirable place to configure logging - unless I've misunderstood how these are intended to evolve - does that match your understanding?

**Comment by sungam3r** [comment]
Yes, but it will be only in netcoreapp3.0 and netstandard2.1. Still need to wait for the release. For netstandard2.0 there is no such unification.

**Comment by nblumhardt** [comment]
Thanks for the reply. It sounds like there's a bit of a compromise necessary for the time being on `netcoreapp2.x` 🤔

**Comment by nblumhardt** [comment]
3.0 is now available, I think that's the way forwards. Cheers!

**Pull request closed without merging**